pkg/controllers/workspace_gc_finalizer.go

```diff
@@ -30,6 +32,22 @@ func (c *WorkspaceReconciler) garbageCollectWorkspace(ctx context.Context, wObj
 		}
 	}
 
+	if c.KubernetesVersion.Major >= "29" {
```

Comment:
guard with featuregate. We suppose to remove hardcode 29?

Reply:
Addressed.